File: Ej2/src/bicicleta.py
from abc import ABC


class Bicicleta(ABC):

    def __init__(self, identificador, tipo):
        self._identificador = identificador
        self._tipo = tipo

    def get_identificador(self):
        return self._identificador

    def get_tipo(self):
        return self._tipo

    def duplicar(self):
        return Bicicleta(self._identificador, self._tipo)

File: Ej1/src/practica1ds/Bicicleta.java
package practica1ds;

public abstract class Bicicleta {

    private String tipo;
    private String identificador;

    public Bicicleta(String tipo, String identificador) {
        this.tipo = tipo;
        this.identificador = identificador;
    }

    public String getTipo() {
        return tipo;
    }

    public String getIdentificador() {
        return identificador;
    }

    public String toString() {
        return "Bicicleta " + tipo + " " + identificador;
    }
}